src/Debug.h

```diff
@@ -22,16 +22,9 @@
 #if defined(assert)
 #undef assert
 #endif
-
-#if PURIFY
-#define assert(EX) ((void)0)
-#elif defined(NODEBUG)
-#define assert(EX) ((void)0)
-#elif STDC_HEADERS
-#define assert(EX)  ((EX)?((void)0):xassert( # EX , __FILE__, __LINE__))
-#else
-#define assert(EX)  ((EX)?((void)0):xassert("EX", __FILE__, __LINE__))
-#endif
+// re-do custom assert definitions
+#undef SQUID_ASSERT_H
+#include "compat/assert.h"
```

Comment:
_Why_ do we want to force a second `compat/assert.h` inclusion here? If this is done because some earlier system `#include` can redefine our own `assert` macro, then we have a big problem -- the same assert() redefinition can happen in other Squid sources that include system files _after_ this Debug.h.

N.B. The inclusion of assert definition in Debug.h is itself a (small) bug because Debug.h does not call assert() and, to a lesser extent, also because 60+ sources with assert() callers do not call debugs(). I do not know whether fixing that bug is in this PR scope because I have not studied why these lines were modified in this PR in the first place.

Reply:
IIRC One of the iomanip or sstream include dependencies has build issues if assert() already exists. So we had to temporarily disable then rebuild our definition. You are welcome to try and fix it, but the issue and logic existed before this PR so it out of scope here.

Reply:
> IIRC One of the iomanip or sstream include dependencies has build issues if assert() already exists

Would not that imply that folks cannot include `<cassert>` before `<iomanip>` in their projects? I bet there are no such restrictions in supported environments. FWIW, I see no assert() use in GCC v9 and v10 STL headers (except for cassert-related declarations, of course).

If you do not want to investigate (so that we can make appropriate PR modifications without speculation), then let's not change these lines (beyond the absolute minimum necessary to stay in sync with in-scope configure.ac changes). I certainly do not insist on any assert-related investigations in this unrelated PR! Adding a TODO to mark this existing assert-defining code duplication is fine, of course, but even that is optional IMO.

We should not attempt to improve this code if initial improvement attempts result in an even stranger-looking code that we cannot fully explain while proper improvement requires investigations that we do not want to perform.

Reply:
reverted in 67eafc2